[log4net] update docs (#4022)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/log4net-context-injection.md

Reviewer: Shouldn't we refer to this doc somewhere? E.g. in `config.md`?

# `log4net` trace context injection

> [!IMPORTANT]
> log4net trace context injection is an experimental feature.

The `log4net` trace context injection is enabled by default. It can be disabled by setting `OTEL_DOTNET_AUTO_LOGS_LOG4NET_INSTRUMENTATION_ENABLED` to `false`.

Context injection is supported for `log4net` in versions >= 2.0.13 && < 4.0.0

> Following properties are added by default to the collection of logging event's properties:

Author: ```suggestion
Following properties are set by default on the collection of logging event's properties:
```
